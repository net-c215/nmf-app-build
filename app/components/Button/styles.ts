import { StyleSheet } from "react-native";
import colors from "../../style/colors";

export default StyleSheet.create({
  default: {
    paddingHorizontal: 28
  },
  black: {
    backgroundColor: colors.darkLink
  },
  fullWidth: {
    width: "100%",
    alignItems: "center",
    paddingHorizontal: 0
  }
});
