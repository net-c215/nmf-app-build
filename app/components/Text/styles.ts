import { StyleSheet } from "react-native";
import { Colors, Font } from "../../style";

export default StyleSheet.create({
  red: {
    color: Colors.bloodyRed,
  },
  darkGray: {
    color: Colors.darkLink70,
  },
  lightGray: {
    color: Colors.swordGray,
  },
  green: {
    color: Colors.linkGreen,
  },
  white: {
    color: Colors.white,
  },
  bold: {
    fontFamily: Font.FontWeight.Bold,
  },
  lightWeightText: {
    fontFamily: Font.FontWeight.Light,
  },
  center: {
    textAlign: "center",
  },
});
