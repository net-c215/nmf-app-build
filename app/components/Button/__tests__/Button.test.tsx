import React from "react";
import renderer from "react-test-renderer";
import Button from "../";
import Text from "../../Text";

jest.unmock("../");

/* Primary Button Tests */

it("renders correctly full width Primary Button with Primary font", () => {
  const tree = renderer
    .create(
      <Button.Primary fullWidth onPress={() => {}} textType={"Primary"}>
        <Text.Primary>test</Text.Primary>
      </Button.Primary>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders correctly Primary Button with Secondary font", () => {
  const tree = renderer
    .create(
      <Button.Primary onPress={() => {}} textType={"Secondary"}>
        <Text.Secondary>test</Text.Secondary>
      </Button.Primary>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders correctly Primary Button with Tertiary font", () => {
  const tree = renderer
    .create(
      <Button.Primary onPress={() => {}} textType={"Tertiary"}>
        <Text.Tertiary>test</Text.Tertiary>
      </Button.Primary>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

/* Secondary Button Tests */

it("renders correctly full width Secondary Button with Primary font", () => {
  const tree = renderer
    .create(
      <Button.Secondary fullWidth onPress={() => {}} textType={"Primary"}>
        <Text.Primary>test</Text.Primary>
      </Button.Secondary>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders correctly Secondary Button with Secondary font", () => {
  const tree = renderer
    .create(
      <Button.Secondary onPress={() => {}} textType={"Secondary"}>
        <Text.Secondary>test</Text.Secondary>
      </Button.Secondary>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders correctly Secondary Button with Tertiary font", () => {
  const tree = renderer
    .create(
      <Button.Secondary onPress={() => {}} textType={"Tertiary"}>
        <Text.Tertiary>test</Text.Tertiary>
      </Button.Secondary>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
