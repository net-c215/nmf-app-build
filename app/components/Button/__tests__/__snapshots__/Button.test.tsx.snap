// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly Primary Button with Secondary font 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#4BB377",
      "borderRadius": 40,
      "opacity": 1,
      "paddingHorizontal": 28,
      "paddingVertical": 12,
    }
  }
>
  <Text.Secondary>
    test
  </Text.Secondary>
</View>
`;

exports[`renders correctly Primary Button with Tertiary font 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#4BB377",
      "borderRadius": 38,
      "opacity": 1,
      "paddingHorizontal": 28,
      "paddingVertical": 12,
    }
  }
>
  <Text.Tertiary>
    test
  </Text.Tertiary>
</View>
`;

exports[`renders correctly Secondary Button with Secondary font 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "borderColor": "#4BB377",
      "borderRadius": 40,
      "borderWidth": 2.5,
      "opacity": 1,
      "paddingHorizontal": 28,
      "paddingVertical": 12,
    }
  }
>
  <Text.Secondary>
    test
  </Text.Secondary>
</View>
`;

exports[`renders correctly Secondary Button with Tertiary font 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "borderColor": "#4BB377",
      "borderRadius": 38,
      "borderWidth": 2.5,
      "opacity": 1,
      "paddingHorizontal": 28,
      "paddingVertical": 12,
    }
  }
>
  <Text.Tertiary>
    test
  </Text.Tertiary>
</View>
`;

exports[`renders correctly full width Primary Button with Primary font 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#4BB377",
      "borderRadius": 42,
      "opacity": 1,
      "paddingHorizontal": 0,
      "paddingVertical": 12,
      "width": "100%",
    }
  }
>
  <Text.Primary>
    test
  </Text.Primary>
</View>
`;

exports[`renders correctly full width Secondary Button with Primary font 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#FFFFFF",
      "borderColor": "#4BB377",
      "borderRadius": 42,
      "borderWidth": 2.5,
      "opacity": 1,
      "paddingHorizontal": 0,
      "paddingVertical": 12,
      "width": "100%",
    }
  }
>
  <Text.Primary>
    test
  </Text.Primary>
</View>
`;
